src/Evolution/Systems/Burgers/FiniteDifference/BoundaryGhostData.hpp

```diff
@@ -0,0 +1,221 @@
+                                    tmpl::_1>>>;
+
+      // Now apply subcell boundary conditions
+      call_with_dynamic_type<void, derived_boundary_conditions_for_subcell>(
+          &boundary_condition, [&boundary_condition, &box, &direction,
+                                &ghost_data_vars, &num_face_pts, &subcell_mesh](
+                                   const auto& derived_boundary_condition_v) {
+            using BoundaryCondition =
+                std::decay_t<decltype(*derived_boundary_condition_v)>;
+            if (dynamic_cast<const BoundaryCondition*>(&boundary_condition) !=
+                nullptr) {
+              using bcondition_interior_evolved_vars_tags =
+                  typename BoundaryCondition::
+                      fd_interior_evolved_variables_tags;
+              using bcondition_interior_temporary_tags =
+                  typename BoundaryCondition::fd_interior_temporary_tags;
+              using bcondition_gridless_tags =
+                  typename BoundaryCondition::fd_gridless_tags;
+
+              using bcondition_interior_tags =
+                  tmpl::append<bcondition_interior_evolved_vars_tags,
+                               bcondition_interior_temporary_tags,
+                               bcondition_gridless_tags>;
+
+              if constexpr (BoundaryCondition::bc_type ==
+                            evolution::BoundaryConditions::Type::Ghost) {
+                // Note: `direction` is required by DirichletAnalytic BC (not
+                // from Dirichlet)
+                auto apply_fd_ghost =
+                    [&boundary_condition, &direction, &ghost_data_vars](
+                        const auto&... boundary_ghost_data_args) {
+                      dynamic_cast<const BoundaryCondition&>(boundary_condition)
+                          .fd_ghost(make_not_null(&get<Burgers::Tags::U>(
+                                        ghost_data_vars)),
+                                    direction, boundary_ghost_data_args...);
+                    };
+                apply_subcell_boundary_condition_impl(
+                    apply_fd_ghost, box, bcondition_interior_tags{});
+              } else if constexpr (BoundaryCondition::bc_type ==
+                                   evolution::BoundaryConditions::Type::
+                                       Outflow) {
+                // Outflow boundary condition only checks if the characteristic
+                // speed is directed out of the element.
+                const auto& volume_mesh_velocity =
+                    db::get<domain::Tags::MeshVelocity<1, Frame::Inertial>>(
+                        *box);
+                std::optional<tnsr::I<DataVector, 1>> face_mesh_velocity{};
+                if (volume_mesh_velocity.has_value()) {
+                  face_mesh_velocity = tnsr::I<DataVector, 1>{num_face_pts};
+                  evolution::dg::project_tensor_to_boundary(
```

Comment:
ping

Reply:
I copied these from the line 270-275 of `Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp`.. I thought it might not be an issue if we are just projecting subcell volume mesh to subcell face mesh?

Reply:
I copied these from the line 270-275 of `Evolution/DiscontinuousGalerkin/Actions/BoundaryConditionsImpl.hpp` at the first place, but now I see there's a problem as you pointed out

Reply:
Maybe here we need another free function to be written in a separate PR, which gets functions of time, logical to grid map, .... etc. so it computes volume mesh velocity on the extended (volume + ghost) subcell grid points so that we can reconstruct face mesh velocity on the boundary interface..?

Reply:
Yea, but for now just do `ERROR` because there are _a lot_ of things that need to be tested once we do subcell with a moving mesh :) This shouldn't hold up the PR :)